### PR TITLE
Simplify Tuplizer

### PR DESCRIPTION
**Pull request by anmalara**
[ ci skip]

- [x] Storage class: now derived classed inherit from the base class, add printing option, add storage T2ULB
- [x] Sample class:
    - [x] expanded years, but clear unused ones
    - [x] added printing options
    - [x] added VerifyStage macro
    - [x] standardized default output of functions
    - [x] simplified code 
    - [x] Added SampleContainer class
    - [x] updated get_missing_tuples function
- [x] Fix a few typos
- [x] Renamed findMissingFilesT3 -> findMissingRootFiles, removed generation_step as not used and substituted with treename
- [x] Updated naming convention for years: EOY/UL+16(pre/post)-17-18
- [x] Removed unused files
- [x] fixed import within classes
- [x] moved to safe_load: couldn't be bothered to deal with the proper file ending
- [x] Created macro to get dasgo client options
- [x] removed unused libraries
- [x] Simplified TuplizeRunner class init: variables taken from env
- [x] Added HT condor submission for the Tuplizer class
- [x] Check includes
- [x] Check default paths
- [x] check __init__
- [x] check sample class attributes

Do not merge yet

### CODE REVIEW

File: Tuplizer/TuplizeRunner.py [hunk 1]
```diff
@@ -57,28 +38,26 @@ def SubmitTuplize(self, ncores=1, runtime=(01,00), nevt_per_job=200000, mode='ne
...
         commands = []
         njobs = 0
-        for filename in filedict:
-            nevt_thisfile = filedict[filename]
+        for filename, nevt_thisfile in filedict.items():
             njobs_thisfile = int(math.ceil(float(nevt_thisfile)/nevt_per_job))
             for n in range(njobs_thisfile):
-                outfilename = 'NTuples_%s_%i.root' % (stagetag, njobs+1)
+                outfilename = outfoldername+'NTuples_%s_%i.root' % (stagetag, njobs+1)
                 command = 'cmsRun %s type=%s infilename=%s outfilename=%s idxStart=%i idxStop=%i year=%s' % (os.path.join(self.macrofolder, 'python', 'ntuplizer_cfg.py'), self.sample.type, filename, outfilename, n*nevt_per_job, (n+1)*nevt_per_job, self.year)
```
Comment (reimersa) [inline Tuplizer/TuplizeRunner.py:61]: make `macropath` get ANALYZERPATH from env

File: Tuplizer/TuplizeRunner.py [hunk 2]
```diff
@@ -57,28 +38,26 @@ def SubmitTuplize(self, ncores=1, runtime=(01,00), nevt_per_job=200000, mode='ne
...
-            stagetag = 'MINIAOD'
         commands = []
         njobs = 0
-        for filename in filedict:
-            nevt_thisfile = filedict[filename]
+        for filename, nevt_thisfile in filedict.items():
             njobs_thisfile = int(math.ceil(float(nevt_thisfile)/nevt_per_job))
             for n in range(njobs_thisfile):
-                outfilename = 'NTuples_%s_%i.root' % (stagetag, njobs+1)
+                outfilename = outfoldername+'NTuples_%s_%i.root' % (stagetag, njobs+1)
```
Comment (reimersa) [inline Tuplizer/TuplizeRunner.py:60]: remove `outfoldername+`

File: Tuplizer/TuplizeRunner.py [hunk 3]
```diff
@@ -87,70 +66,88 @@ def SubmitTuplize(self, ncores=1, runtime=(01,00), nevt_per_job=200000, mode='ne
...
-            if njobs > 0:
-                if self.submit:
-                    jobid = int(subprocess.check_output(command, shell=True))
-                    print green("  --> Submitted an array of %i jobs for name %s with jobid %s"%(njobs_per_array[narrays], samplename, jobid))
-                else:
-                    print command
-                    print yellow("  --> Would submit an array of %i jobs for name %s"%(njobs_per_array[narrays], samplename))
-            njobs_left -= arrayend
-            narrays += 1
+        if 'slurm' in self.cluster.lower():
```
Comment (reimersa) [inline Tuplizer/TuplizeRunner.py:87]: Here and later: Think about making python code much more similar for all cluster cases, and put remaining differences into `.sh` script to be submitted to Condor or SLURM

Comment (reimersa) [reply Tuplizer/TuplizeRunner.py:87]: well... maybe not really useful

**Review by reimersa — CHANGES_REQUESTED** [review]
Super great changes and a lot of work, thanks a lot!

File: Tuplizer/GetDasNevents.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+def GetDasNevents(dataset, options):
+    url = 'https://cmsweb.cern.ch/dbs/prod/global/DBSReader'
+    api = DbsApi(url = url)
+    outputDataSets = api.listDatasets(dataset = dataset)
+    for dataset in outputDataSets:
+        inp = dataset['dataset']
+        reply =  api.listBlockSummaries(dataset = inp)
+        reply = dict(filter(lambda elem: elem[0] in options, reply[0].items()))
+        print (inp, reply)
```
[the remark this answers is not in the document]
Comment (anmalara) [reply Tuplizer/GetDasNevents.py:21]: done

File: Samples/Sample.py [hunk 5]
```diff
@@ -1,57 +1,50 @@
...
+            raise AttributeError('Invalid key for year-dependent object. Given keys: '+(', '.join(vals.keys()))+'. Supported keys: '+(', '.join(self.__dict.keys())))
+        self.__dict.update(vals)
 
-    def __getitem__(self, year):
-        if self.has_year(year):
-            return self.dict[year]
-        else:
-            return None
+        for key, val in self.__dict.items():
+            if val==None:
```
[the remark this answers is not in the document]
Comment (anmalara) [reply Samples/Sample.py:25]: done

File: Tuplizer/TuplizeRunner.py (hunk 1 above)
Comment (anmalara) [reply Tuplizer/TuplizeRunner.py:61]: done

File: Tuplizer/TuplizeRunner.py (hunk 2 above)
Comment (anmalara) [reply Tuplizer/TuplizeRunner.py:60]: done

File: Tuplizer/TuplizeRunner.py (hunk 3 above)
Comment (anmalara) [reply Tuplizer/TuplizeRunner.py:87]: done

**Pull request merged**